src/AspNet.Security.OAuth.Weibo/WeiboAuthenticationHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections.Generic;
+using System.Net.Http;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http.Authentication;
+using System.Security.Claims;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using AspNet.Security.OAuth.Extensions;
+using Newtonsoft.Json.Linq;
+using JetBrains.Annotations;
+
+namespace AspNet.Security.OAuth.Weibo
+{
+    public class WeiboAuthenticationHandler : OAuthHandler<WeiboAuthenticationOptions>
+    {
+        public WeiboAuthenticationHandler([NotNull] HttpClient client) : base(client) { }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity, [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens)
+        {
+            var queryString = new Dictionary<string, string>()
+            {
+                {"access_token",tokens.AccessToken },
+                {"uid",tokens.Response.Value<string>("uid") }
+            };
+            var endpoint = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, queryString);
+            var response = await Backchannel.GetAsync(endpoint, Context.RequestAborted);
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError($"An error occurred while retrieving the user information: the remote server returned a " +
```

Comment:
This doesn't match what the other providers do. Please make everything consistent.